packages/bng-metric-service/src/helpers/extractors/bng-metric-single-data-extractor.js

```diff
@@ -66,75 +67,47 @@ class BngMetricSingleDataExtractor {
         })
         data = resultData
       } else {
-        data = this.#performSubstitution(data, extractionConfiguration)
-        data = this.#removeUnwantedColumns(data, extractionConfiguration)
+        data = data.reduce((newData, item) => {
+          // #1 Checking each row's to verify if all coliumn has valid values
+          const hasValidValuesInRow = Object.keys(item).some(key => key !== 'a' && !_.isEmpty(item[key]))
+
+          // #2 Preparing obejct on the basis of 'extractionConfiguration'
+          if (hasValidValuesInRow) {
+            item = this.#prepareDataItems(item, extractionConfiguration)
+            newData.push(item)
+          }
+          return newData
+        }, [])
+        // #3 Removing unwanted rows based on the config - rowsToBeRemovedTemplate
```

Comment:
Following review of this pull request @paulsimonandrews  and I have agreed that metric extraction optimisation should be excluded from this story. The proposed optimisation changes appear to stop the extraction of a small subset of data and the implications of doing this need to be analysed after MVP is delivered, Please revert all changes to this file and refactor the associated test changes accordingly.